### ISSUE TITLE
Cannot run setuid commands like sudo with spawn

### ISSUE THREAD

**Issue by cchiu1** [pexpect/pexpect#104]
I installed the pexpect on Python 2.7.7 and CentOS 6.5.
On a terminal, I can run "sudo ls".
However, if I put "pexpect.spawn('sudo ls')" in a Python script, I always get the exception "The command was not found or was not executable: sudo.".
I confirmed that "sudo" is in /usr/bin/.
How do I make it executable for "pexpect"?

**Comment by egrim** [comment]
Here's an alternative fix to get `sudo` working for anyone else running into this:
```python
pexpect.which, original_which = (lambda filename: '/usr/bin/sudo' if filename in ('sudo', '/usr/bin/sudo') else original_which(filename),
                                 pexpect.which)
```

**Comment by jquast** [comment]
Fixed in upcoming 4.0 release by #106. #106 was earmarked for release 3.4 which was meant to be release but did not happen for some unfortunate reason, submitted PR #257 to avoid such release confusion.

Hope to have 4.0 out soon, just awaiting @takluyver's approval of pending PR's and his publish ok.
